src/waylandhelper.cpp

```diff
@@ -0,0 +1,46 @@
+#include "waylandhelper.h"
+
+WaylandHelper::WaylandHelper(QWindow *parent)
+    : QObject(parent)
+    , surface(Surface::fromWindow(parent))
+    , registry(new Registry(this))
+    , connection(ConnectionThread::fromApplication(this))
+{
+    if (!surface->isValid())
+        qFatal("Invalid Wayland surface");
+    registry.create(connection);
+    registry.setup();
+    if (!registry.isValid())
+        qFatal("Invalid Wayland registry");
+    connect(connection, &ConnectionThread::connectionDied, &registry, &Registry::destroy);
+    connect(&registry, &Registry::exporterUnstableV2Announced, this, &WaylandHelper::onXdgExporterAnnounced);
+    connect(&registry, &Registry::interfacesAnnounced, this, &WaylandHelper::onInterfacesAnnounced);
+}
+
+void WaylandHelper::onXdgExporterAnnounced(uint name, uint version)
+{
+    xdgExporter = registry.createXdgExporter(name, version, this);
+    if (!xdgExporter->isValid())
+        qFatal("Invalid Wayland XdgExporter");
+    xdgExportedTopLevel = xdgExporter->exportTopLevel(surface, this);
+    if (!xdgExportedTopLevel->isValid())
+        qFatal("Invalid Wayland toplevel export");
+    connect(xdgExportedTopLevel, &XdgExported::done, [=] {
+        emit doneXdgExport();
+    });
```

Comment:
You can connect signals to other signals directly:
```suggestion
    connect(xdgExportedTopLevel, &XdgExported::done, this, &WaylandHelper::doneXdgExport);
```

Reply:
Neat.